include/astshim/ChebyMap.h

```diff
@@ -0,0 +1,321 @@
+        values are rounded to the nearest integer.
+
+    For instance, if the @ref ChebyMap has 3 inputs and 2 outputs, each row consisting
+    of 5 elements, and the row `(1.2, 2, 6, 3, 0)` describes a coefficient
+    that increments output 2 as follows:
+        `out2 += 1.2 * T6(in1') * T3(in2') * T0(in3')
+    and the row `(-1.5, 1, 0, 0, 0)` describes a coefficient that increments
+    output 2 with a constant value of -1.5 (since all powers are 0):
+        `out1 += -1.5 * T0(in1') * T0(in2') * T0(in3')
+    where inI' is the normalized value of input axis I
+    Each final output coordinate value is the sum of the terms described
+    by the `ncoeff_f` columns in the supplied array. If no coefficients increment
+    a given output then it will be set to 0.
+
+    The coefficients describing the inverse transformation work the same way,
+    but of course the matrix is of size `ncoeff_i x (2 + nout)`, where `ncoeff_i` is the
+    number of coefficients for the inverse transformation.
+    */
+    explicit ChebyMap(ndarray::Array<double, 2, 2> const &coeff_f,
+                      ndarray::Array<double, 2, 2> const &coeff_i, std::vector<double> const &lbnd_f,
+                      std::vector<double> const &ubnd_f, std::vector<double> const &lbnd_i,
+                      std::vector<double> const &ubnd_i, std::string const &options = "")
+            : Mapping(reinterpret_cast<AstMapping *>(
+                      _makeRawChebyMap(coeff_f, coeff_i, lbnd_f, ubnd_f, lbnd_i, ubnd_i, options))) {}
+
+    /**
+    Construct a @ref ChebyMap with only the forward transform specified.
+
+    If the polynomial is invertible and you want an inverse can you call polyTran to fit one
+    (at this time the iterative inverse offered by PolyMap is not available for ChebyMap).
+
+    @param[in] coeff_f  A `(2 + nin) x ncoeff_f` @ref ChebyMap_CoefficientMatrices "matrix of coefficients"
+        describing the forward transformation.
+    @param[in] nout  Number of output coordinates.
+    @param[in] lbnd_f  Lower bounds for input data; one element per input axis
+    @param[in] ubnd_f  Upper bounds for input data; one element per input axis
+    @param[in] options  Comma-separated list of attribute assignments.
+
+    @throw std::invalid_argument if the forward transform is not defined (coeff_f is empty)
+    @throw std::invalid_argument if lbnd_f or ubnd_f do not have nin elements
+    */
+    explicit ChebyMap(ndarray::Array<double, 2, 2> const &coeff_f, int nout,
+                      std::vector<double> const &lbnd_f, std::vector<double> const &ubnd_f,
+                      std::string const &options = "IterInverse=0")
+            : Mapping(reinterpret_cast<AstMapping *>(
+                      _makeRawChebyMap(coeff_f, nout, lbnd_f, ubnd_f, options))) {}
+
+    virtual ~ChebyMap() {}
+
+    ChebyMap(ChebyMap const &) = delete;
+    ChebyMap(ChebyMap &&) = default;
```

Comment:
I want to return a `ChebyMap`, not a shared pointer. `copy` returns a shared pointer.